docker-compose.yml

```diff
@@ -0,0 +1,79 @@
+version: "3.8"
+services:
+  db:
+    image: postgres:14-alpine
+    networks:
+      new:
+        aliases:
+          - database
+    env_file:
+      - .env
+    ports:
+      - ${POSTGRES_PORT}:${DOCKER_INTERNAL_PORT}
+    healthcheck:
+      test:
+        [
+          "CMD",
+          "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}"
+        ]
+      interval: 10s
+      timeout: 3s
+      retries: 5
+    volumes:
+      - pgdata:/var/lib/postgresql/findadoc-data # persist data even if container shuts down
+  db-configure:
+    image: postgres:14-alpine
+    depends_on:
+      - db
+    networks:
+      new:
+        aliases:
+          - database
+    env_file:
+      - .env
+    environment:
+      - DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${DOCKER_POSTGRES_HOST}:${DOCKER_INTERNAL_PORT}/${POSTGRES_DB}?schema=public
+      - SHADOW_DB_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${DOCKER_POSTGRES_HOST}:${DOCKER_INTERNAL_PORT}/${SHADOW_DB}?schema=public
+    entrypoint: /docker/db-init.sh
+    volumes:
+      - ./docker:/docker
+      - status:/out
+  install-deps:
+    depends_on:
+      - db-configure
+    env_file:
+      - .env
+    image: node:18.12.1-alpine
+    networks:
+      - new
+    working_dir: /target
+    volumes:
+      - .:/target
+    command: "yarn install"
+  migrate:
+    depends_on:
+      - db-configure
+      - install-deps
+    env_file:
+      - .env
+    environment:
+      - DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${DOCKER_POSTGRES_HOST}:${DOCKER_INTERNAL_PORT}/${POSTGRES_DB}?schema=public
+      - SHADOW_DB_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${DOCKER_POSTGRES_HOST}:${DOCKER_INTERNAL_PORT}/${SHADOW_DB}?schema=public
+    links:
+      - db
+    networks:
+      - new
+    image: node:18.12.1-alpine
+    working_dir: /target
+    volumes:
+      - .:/target
+      - status:/in
+    command: "sh -c 'until [[ -f /in/db-init-done && -d /target/node_modules ]]; do sleep 1; done; yarn migrate; yarn prisma db seed'"
```

Comment:
idea: it seems like we're doing a fair amount of setup in the compose script. It might be a little cleaner to build a wrapper docker image around the `postgres:14-alpine` image and move things like this command inside the container. 
This would lessen our dependency on docker compose and make the move easier to something like AWS container services easier in the future.

Reply:
non-blocking question: ^ after I was looking at the different services in the compose file, should `db-configure`, `install-deps`, and `migrate` all be tasks under a singular `db` docker file? 

I was imagining this compose file is going to managing our node service, db, etc and everything for the findadoc backend. Is this what you're thinking or rather a separate compose file for the db and node services?

Reply:
Honestly this took several weekends to get working and I don't want to waste energy optimizing it when it works. Fine for folks to tune up later